eclair-core/src/main/scala/fr/acinq/eclair/package.scala

```diff
@@ -30,11 +31,24 @@ package object eclair {
    * We are using 'new SecureRandom()' instead of 'SecureRandom.getInstanceStrong()' because the latter can hang on Linux
    * See http://bugs.java.com/view_bug.do?bug_id=6521844 and https://tersesystems.com/2015/12/17/the-right-way-to-use-securerandom/
    */
-  val secureRandom = new SecureRandom()
+  private val secureRandom = new SecureRandom()
+
+  def initRandom(): Unit = secureRandom.nextInt()
+
+  /**
+   * We're using an additional, weaker randomness source to protect against catastrophic failures of the SecureRandom
+   * instance.
+   */
+  val weakRandom = new WeakRandom()
```

Comment:
We should find a way to not make it possible to access `weakRandom.nextBytes` so easily. I understand that `weakRandom.addEntropy()` needs to stay public.

Reply:
Easy, I can export different traits for each purpose

Reply:
Done in https://github.com/ACINQ/eclair/pull/1774/commits/0df44d9d5a3f6beac430640e213f8603185f57f0